backend/internal/flow/store/store.go

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package store provides the implementation for flow context persistence operations.
+package store
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/asgardeo/thunder/internal/flow/model"
+	"github.com/asgardeo/thunder/internal/system/database/provider"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const loggerComponentName = "FlowStore"
+
+// StoreFlowContext stores the complete flow context in the database.
+func StoreFlowContext(ctx model.EngineContext) error {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("runtime")
```

Comment:
Shall we define a struct for the FlowStore and declare the DBProvider within it? We can then define the provider while creating the instance? 

This may be the current approach in most of the places, however it doesn't allow mocking the database client for unit tests